library/Makefile

```diff
@@ -35,16 +37,26 @@ SOEXT_TLS=so.10
 SOEXT_X509=so.0
 SOEXT_CRYPTO=so.2
 
-# Set DLEXT=dylib to compile as a shared library for Mac OS X
 DLEXT ?= so
 
 # Set AR_DASH= (empty string) to use an ar implentation that does not accept
 # the - prefix for command line options (e.g. llvm-ar)
 AR_DASH ?= -
 
-# Windows shared library extension:
+ifdef APPLE_BUILD
+ARFLAGS = $(AR_DASH)Src
```

Comment:
Why are these flags dependent on the platform and platform alone. These flags can change with compiler. If they change for a platform  and compiler selection then both should be checked. Have tested that this change will suit all compilers?

Reply:
Due to peculiarities of macOS, it's really rare to use the non-system compiler. It's possible, but requires going out to Mac Homebrew or other 3rd party packages, or building the compiler yourself from source. Due to how rare using other compilers is, we should deal with any issues in using those compilers with our Makefile as separate bugs and PRs to fix those bugs.